kpi/management/commands/sync_form_media.py

```diff
@@ -0,0 +1,194 @@
+from django.core.management.base import BaseCommand
+
+from rest_framework import status
+from rest_framework.authtoken.models import Token
+
+from kpi.models import Asset, AssetFile
+
+
+URL_MEDIA_LIST = '{kc_url}/api/v1/metadata?data_type=media&format=json'
+URL_MEDIA_CONTENT = '{kc_url}/{username}/forms/{asset_uid}/formid-media/{media_id}'
+
+
+def _get_asset_metadata(token: str) -> Optional[dict]:
+    url = URL_MEDIA_LIST.format(kc_url=settings.KOBOCAT_INTERNAL_URL)
+    response = requests.get(
+        url=url, headers={'Authorization': f'Token {token}'}
+    )
+    if response.status_code == status.HTTP_200_OK:
+        return response.json()
+
+
+def _get_media_file_content(
+    token: str,
+    username: str,
+    asset_uid: str,
+    media_id: str,
+) -> bytes:
+    url = URL_MEDIA_CONTENT.format(
+        kc_url=settings.KOBOCAT_INTERNAL_URL,
+        username=username,
+        asset_uid=asset_uid,
+        media_id=media_id,
+    )
+    response = requests.get(
+        url=url, headers={'Authorization': f'Token {token}'}
+    )
+    return response.content
+
+
+def _sync_media_files(
+    asset_uid: Optional[str],
+    username: Optional[str],
+    verbosity: int,
+    *args,
+    **kwargs
+) -> dict:
+
+    assets = Asset.objects.filter(_deployment_data__isnull=False)
+    assets_all_count = assets.count()
+
+    if username is not None:
+        assets = assets.filter(owner__username=username)
+    if assets and asset_uid is not None:
+        assets = assets.filter(uid=asset_uid)
+
+    assets_selected_count = assets.count()
+
+    sync_stats_all = []
+    for asset in assets:
+        sync_stats = {}
+        if not asset.has_deployment:
+            continue
+        asset_xform_id = asset.deployment.backend_response['formid']
+        user = asset.owner
+        token = Token.objects.get(user=user)
+
+        # for logging stats
+        kpi_before = AssetFile.objects.filter(
+            asset=asset,
+            file_type=AssetFile.FORM_MEDIA,
+            date_deleted__isnull=True,
+        ).count()
+
+        media_files = _get_asset_metadata(token)
+        if media_files is None:
+            continue
+
+        synced_files = []
+        for media_file in media_files:
+            media_filename = media_file['data_value']
+            media_id = media_file['id']
+            media_xform_id = media_file['xform']
+
+            # Ensure that we send the form-media to correct asset and
+            # don't create duplicates
+            if (
+                media_xform_id != asset_xform_id
+                or AssetFile.objects.filter(
+                    asset=asset,
+                    date_deleted__isnull=True,
+                    metadata__filename=media_filename,
+                ).exists()
+            ):
+                continue
+
+            media_content = _get_media_file_content(
+                token=token,
+                username=user.username,
+                asset_uid=asset.uid,
```

Comment:
It's possible that `KPI.Asset.uid` can be different than `KoBoCAT.XForm.id_string`.
This makes me bring my comment about `XForm.id_string` vs `Asset.uid` ;-)

I think it should be rewritten this way to support projects uploaded to KC directly.

```python
            media_content = _get_media_file_content(
                token=token,
                username=user.username,
                xform_id_string=asset.deployment.xform_id_string,
                media_id=media_id,
            )
```